waspc/src/Wasp/Generator/ServerGenerator/AuthG.hs

```diff
@@ -22,11 +38,13 @@ genAuth spec = case maybeAuth of
       [ genCoreAuth auth,
         genAuthMiddleware auth,
         -- Auth routes
-        genAuthRoutesIndex,
+        genAuthRoutesIndex auth,
         genLoginRoute auth,
         genSignupRoute auth,
-        genMeRoute auth
+        genMeRoute auth,
+        genUtilsJs auth
       ]
+      <++> (if AS.Auth.isExternalAuthEnabled auth then genPassportAuth auth else return [])
```

Comment:
Maybe we should put this check inside `genPassportAuth` to stay consistent with the rest of the code (for example, this function can return `Generator []` if `maybeAuth` ends up being false).

Reply:
Good points on these 3 instances. I think I did have them inside the function before, but Martin and I were discussing where they make the most sense. The pro for here is we make it explicit what the condition for inclusion is, and it cleans up the function some. But the con is do we even need to care, or just trust the function to do the right thing and abstract the check away from the call site? I think I'll go back to your suggestion here, if nothing else for consistency with the existing code as you mentioned. 👍🏻